homeassistant/components/onvif/__init__.py

```diff
@@ -123,37 +121,23 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     )
 
 
-async def _get_snapshot_auth(hass, device, entry):
+async def _get_snapshot_auth(device):
     if not (device.username and device.password):
         return HTTP_DIGEST_AUTHENTICATION
 
-    snapshot_uri = await device.async_get_snapshot_uri(device.profiles[0])
-    if not snapshot_uri:
-        return HTTP_DIGEST_AUTHENTICATION
-    auth = HTTPDigestAuth(device.username, device.password)
-
-    def _get():
-        # so we can handle keyword arguments
-        return requests.get(snapshot_uri, timeout=1, auth=auth)
-
     try:
-        response = await hass.async_add_executor_job(_get)
-
-        if response.status_code == HTTP_UNAUTHORIZED:
-            return HTTP_BASIC_AUTHENTICATION
+        snapshot = await device.device.get_snapshot(device.profiles[0].token)
 
-        return HTTP_DIGEST_AUTHENTICATION
-    except requests.exceptions.Timeout:
+        if snapshot:
+            return HTTP_DIGEST_AUTHENTICATION
+        return HTTP_BASIC_AUTHENTICATION
+    except (ONVIFAuthError, ONVIFError, ONVIFTimeoutError):
```

Comment:
I'm not sure about a general `ONVIFError` - this could lead to intermittent errors having basic instead of digest. My suggestion is to have it in a separate `except` and return `HTTP_DIGEST_AUTHENTICATION`. That's also why I split `ONVIFAuthError` and `ONVIFTimeoutError` into separate classes.